ktest/httprecorder/http_recorder.go

```diff
@@ -0,0 +1,78 @@
+package httprecorder
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+type HTTPRecorder struct {
+	inner http.RoundTripper
+	log   *RequestLog
+}
+
+func NewRecorder(inner http.RoundTripper, log *RequestLog) *HTTPRecorder {
+	rt := &HTTPRecorder{inner: inner, log: log}
+	return rt
+}
+
+func (m *HTTPRecorder) RoundTrip(request *http.Request) (*http.Response, error) {
+	entry := &LogEntry{}
+	entry.Request = Request{
+		Method: request.Method,
+		URL:    request.URL.String(),
+		Header: request.Header,
+	}
+
+	if request.Body != nil {
+		requestBody, err := io.ReadAll(request.Body)
+		if err != nil {
+			panic("failed to read request body")
```

Comment:
Why not return an error value here, since the signature allows it?

Reply:
Good call - done!

Reply:
Great call.  Done